content/rust/concepts/variables/variables.md

```diff
@@ -0,0 +1,66 @@
+---
+Title: 'Variable'
+Description: 'A variable is a location in computer memory used to store data.'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Variables'
+CatalogContent:
+  - 'learn-rust'
+  - 'paths/computer-science'
+---
+
+A **variable** is a fundamental concept in computer programming. It represents a location in computer memory used to store data temporarily. Whenever there's a need to store a piece of data for use in a program, a variable is used. This practice ensures code reusability, as variables can replace the same value in multiple locations within the same code.
+
+## Data types in rust
+
+In the Rust programming language, variables are associated with specific data types. These data types determine:
+
+- The layout and size of the variable in memory.
+- The range of values that can be stored within that memory.
+- The operations that can be conducted on the variable.
+
+## Declaring a Variable
+
+In Rust, variable declaration follows a specific syntax pattern. Here's a basic example:
+
+```pseudo
+let number: i32 = 42;
+```
+
+In the above example,a variable is declared named `number` with an initial value of `42` and it specifies its data type as `i32` (a 32-bit signed integer).
+
```

Comment:
It might be more complete if there's a section that explains the `let` and `mut` keywords before the example

Reply:
ok